python/pyiceberg/table/sorting.py

```diff
@@ -135,3 +140,31 @@ def __str__(self) -> str:
 
 UNSORTED_SORT_ORDER_ID = 0
 UNSORTED_SORT_ORDER = SortOrder(order_id=UNSORTED_SORT_ORDER_ID)
+INITIAL_SORT_ORDER_ID = 1
+
+
+def assign_fresh_sort_order_ids(sort_order: SortOrder, old_schema: Schema, fresh_schema: Schema) -> SortOrder:
+    if sort_order.is_unsorted:
+        return UNSORTED_SORT_ORDER
+
+    fresh_fields = []
+    for field in sort_order.fields:
+        original_field = old_schema.find_field(field.source_id)
+        if original_field is None:
+            raise ValueError(f"Could not find in original schema: {field}")
+        fresh_field = fresh_schema.find_field(original_field.name)
```

Comment:
This can't use `original_field.name` because that's the local name. Instead, I think you need to use `old_schema.find_column_name`

Reply:
Thanks!

Reply:
This still needs to be addressed as well.